PXD-443 Feat/file submit

macintoshpie opened the pull request.

macintoshpie committed:
feat(submitFileTest): create tests for testing file submission and updating

updated config for Ted's qa env

added files for testing submission

feat(steps): sheepdog
added sheepdog api functions

feat(suites/apis): submitFileTests
added functions to actor for testing submitting files

feat(suites/apis): submitFilesTest
created tests for submitting files

feat(suites/apis): submitFilesTest
added assertions to test submitted files has correct attributes

thanh-nguyen-dang commented on steps/sheepdogAPI.js:
```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+let username = process.env.INDEX_USERNAME;
```
can you make the code getting username, password, and creating token in the headers as a util function? Don't repeat the code in different places

thanh-nguyen-dang commented on suites/apis/submitFileTest.js:
```diff
@@ -0,0 +1,172 @@
+Feature('SubmitFileTest');
+
+// Nodes in sorted (hierarchical) order
+const nodes = [
```
it better to give every node a `key`

thanh-nguyen-dang commented on codecept.json:
```diff
@@ -2,7 +2,7 @@
   "helpers": {
     "WebDriverIO": {
-      "url": "https://thanhnd.planx-pla.net",
+      "url": "https://tsummer.planx-pla.net",
```
could you remove `codecept.json` from github

thanh-nguyen-dang commented on suites/apis/submitFileTest.js:
```diff
@@ -0,0 +1,172 @@
+};
+
+Scenario('test submit file', async(I) => {
+  I.submitFile("/api/v0/submission/dev/test/", files.test_file)
```
make program and project as parameters

macintoshpie committed:
refactor(submitFileTest): refactored funcs and util funcs
refactor(steps/utilAPI): getAccessTokenHeader
add util function for getting token header

refactor(steps/sheepdogAPI): getAccessTokenHeader
use util function to get header

refactor(steps/fenceAPI): getAccessTokenHeader
use util function to get token header

refactor(steps/sheepdogAPI): add/deleteNodes
sort nodes by key value before adding/deleting them

refactor(submitFileTest): nodes
added key val to nodes for sorting

refactor(sheepdogAPI): refactoring all funcs

refactor(submitFileTest): all tests
refactor some parts of tests

chore(codeceptJson): added to gitignore

refactor(fenceAPI): use indexAuth util for header

refactor(utilAPI): implement indexAuth header util

refactor(submitFileTest): move node data into separate file

refactor(submitSamples): update node data

refactor(sheepdogAPI): query indexd for submitFile and return bool for deleteFile

refactor(submitFileTest): refactored all tests

thanh-nguyen-dang commented on steps/utilAPI.js:
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+module.exports.getAccessTokenHeader = function() {
```
where did you use username/password?

macintoshpie replied:
From your examples it looks like username/password isn't needed in the header.
You only used the username and passwords in fenceAPI.js: addFileIndices and deleteFileIndices. For seeFileContentEqual you just used the info that I added to the util function.

thanh-nguyen-dang commented on steps/fenceAPI.js:
```diff
@@ -2,12 +2,11 @@
-  'Accept': 'application/json',
-  'Authorization': `bearer ${process.env.ACCESS_TOKEN}`
-};
 let indexAuth = Buffer.from(`${username}:${password}`).toString('base64');
```
check how did username/password are used here. Remove repeated code, update the util function

thanh-nguyen-dang commented on suites/apis/submitFileTest.js:
```diff
@@ -0,0 +1,191 @@
+});
+
+BeforeSuite((I) => {
+  I.addNodes("/api/v0/submission/dev/test/", Object.values(nodes))
```
can it be parameterized?

thanh-nguyen-dang commented on suites/apis/submitFileTest.js:
```diff
@@ -0,0 +1,191 @@
+});
+
+AfterSuite((I) => {
+  I.deleteNodes("/api/v0/submission//dev/test/entities/", Object.values(nodes))
```
can it be parameterized?

thanh-nguyen-dang commented on suites/apis/submitFileTest.js:
```diff
@@ -0,0 +1,191 @@
+  },
+];
+
+const files = {
```
where is the link? could you change the filename to the one in s3 bucket?

thanh-nguyen-dang commented on suites/apis/submitFileTest.js:
```diff
@@ -0,0 +1,191 @@
+Feature('SubmitFileTest');
+
+// Nodes, sorted hierarchically by key
+const nodes = [
```
move nodes to a separated test data file because it is too large to put in a code files.

thanh-nguyen-dang commented on steps/sheepdogAPI.js:
```diff
@@ -0,0 +1,107 @@
+        file.id = new_id;
+        let get_file = `${sheep_endpoint}${file.program}/${file.project}/export?ids=${new_id}&format=json`;
+        // get file from sheepdog to get did
+        return this.sendGetRequest(get_file, accessTokenHeader).then(
```
this code should be clearly split into two functions so that easier to read

macintoshpie committed:
refactor(sheepdogAPI): refactor submit file, add and delete nodes

thanh-nguyen-dang commented on sample_data/submitSamples.json:
```diff
@@ -0,0 +1,140 @@
+{
+  "study": {
+    "key": 1,
+    "program": "dev",
```
we still hard code program/project name here?

macintoshpie replied:
No, that is updated in the other pull request, branch feat/graphqlQuery
